test/ui-testing/interfaces.js

```diff
@@ -0,0 +1,216 @@
+                nightmare
+                    .click(`#orgs-nameOrg-${row}-search-button`)
+                    .wait('#input-organization-search')
+                    .type('#input-organization-search', orgName)
+                    .click('[data-test-search-and-sort-submit]')
+                    .waitUntilNetworkIdle(1000)
+                    .evaluate((name) => {
+                        const nameElements = [...document.querySelectorAll('div[role="gridcell"]')];
+                        const organization = nameElements.find(e => e.textContent === name);
+                        if (!organization) throw new Error(`Could not find the organization ${name}`);
+                        organization.click();
+                    }, orgName)
+                    .waitUntilNetworkIdle(1000)
+                    .then(done)
+                    .catch(done)
+            });
+
+            it(`should assign role: ${org.role}`, done => {
+                nightmare
+                    .wait(`#orgs-role-${row}`)
+                    .type(`#orgs-role-${row}`, org.role)
+                    .evaluate((r, o) => {
+                        const roleElement = document.querySelector(`#orgs-role-${r}`);
+                        const role = roleElement.selectedOptions[0].textContent;
+                        if (role !== o.role) {
+                            throw Error(`Expected role to be ${o.role} but is ${role}`);
+                        }
+                    }, row, org)
+                    .then(done)
+                    .catch(done);
+            });
+
+            it('should create Agreement', done => {
+                nightmare
+                    .click('#clickable-createagreement')
+                    .waitUntilNetworkIdle(2000)
+                    .then(done)
+                    .catch(done);
+            });
+
+            it(`should find "${org.name}" in Organizations list with role ${org.role}`, done => {
+                nightmare
+                    .evaluate((org, name) => {
+                        const rows = [...document.querySelectorAll('[data-test-organizations-org]')].map(e => e.textContent);
+                        const row = rows.find(r => r.indexOf(name) >= 0);
+                        if (!row) {
+                            throw Error(`Could not find row with an org named ${name}`);
+                        }
+                        if (row.indexOf(org.role) < 0) {
+                            throw Error(`Expected row for "${name}" to contain role ${org.role}.`);
+                        }
+                    }, org, orgName)
+                    .then(done)
+                    .catch(done);
+            });
+
+            it(`should find "${interfaceName}" in interface list`, done => {
+                nightmare
+                    .evaluate(interfaceName => {
+                        const nameElements = [...document.querySelectorAll('[data-test-card-body ] div[role="gridcell"]')];
```

Comment:
Can you scope this selector to just inside the interface list of the org? Or just inside the orgs section?

Also, can you add tests for the link, username, and password (and notes if they're making it in)?